### ISSUE TITLE
Form with array input fields validating only first

### ISSUE THREAD

**Issue by blpraveen** [jquery-validation/jquery-validation#1226]
In the form I have many fields with name firstname[] 
<input type="text" id="number" name="firstname[]" >

When I validate using the rule
            rules: {
                'firstname[]': {
                    required: true,
                    minlength: 1,
                },
            }

It shows error message only to the first element.
What is the solution to this problem?

**Comment by relizont** [comment]
In jquery.validate.js, we can find a function named checkForm, we have to modify it as below to fix.

```
checkForm: function() {
this.prepareForm();
for ( var i = 0, elements = (this.currentElements = this.elements()); elements[i]; i++ ) {
if (this.findByName( elements[i].name ).length != undefined && this.findByName( elements[i].name ).length > 1) {
for (var cnt = 0; cnt < this.findByName( elements[i].name ).length; cnt++) {
this.check( this.findByName( elements[i].name )[cnt] );
}
} else {
this.check( elements[i] );
}
}
return this.valid();
}
```

**Comment by abhi143u11** [comment]
@relizont your solution worked for me thanks alot i find it after 3 days of searching thanks very much

**Comment by emmanuelLS** [comment]
Hi, I did applied this script on the jquery.validate.js file and it works as expected. However, there is no error message display

**Comment by vannakpanha** [comment]
@relizont  Very nice solution, It worked for me

**Comment by cybersubtype** [comment]
Thanks @relizont this also worked for having multiple file inputs with same name.

**Comment by kamalcheema6** [comment]
@relizont  : Thanks. it is working.

**Comment by pradeepkchandra** [comment]
@relizont: 
I do not find the method checkForm. in validate.js. People where do I add this method. 
I am trying to validate an array of contact methods. and I am not able to validate the index that I am currently modifying. help appreciated

**Comment by rachinkapoor** [comment]
@jzaefferer,
At one place you say that email regexp should not find admin@localhost as invalid because it is "as per standards".  Fair enough, we accept it.

So "keeping up with the standards", why don't you support a form body that has same 'name' for multiple inputs ? 

It is not invalid to have multiple fields with same name.
Also, jquery addressed the same issue using serializeArray.

**Comment by GeorgeGeorgiev** [comment]
Hi Guys,
I have dealt with this problem just add unique ids for the inputs and the validation will work normally.

Regards,
G.

**Comment by SkywalkingZulu** [comment]
Wait so each id has a unique id and there is an array for each unique id? Therefore it becomes an if statement that only let's you pass through itterating through the array till it finds your value. Is this if then or if while statements?
